designs/2021-break-on-parsing-errors/README.md

```diff
@@ -0,0 +1,185 @@
+- Repo: eslint/eslint
+- Start Date: 2021/1/12
+- RFC PR: https://github.com/eslint/rfcs/pull/75
+- Authors: [A-Katopodis](https://github.com/A-Katopodis)
+
+# (Break on parsing errors)
+
+## Summary
+
+The suggested change is for ESLint tp support an argument that will make the CLI exit with code 2 if any fatal errors are reported. The option will be an opt-in boolean argument called `--fatal-parse-error`.
+
+## Motivation
+
+We met with a couple of cases where we assumed a succesfull run of ESLint in CI enviroments. When ESLint wouldn't be able to read the tsconfig.json or had a wrongly configured source type the error would be a `fatal` one but the exit code would be `1`. There was no distiction between a run with no fatal errors and with fatal errors.
```

Comment:
Can we clarify a bit more the invalid/unexisting `tsconfig.json` scenario.

`@typescript-eslint/parser` throws, ESLint treats that as a parsing error and makes a linting message with `fatal: true`, then continues to lint and at the end returns exit code `1`?

Reply:
Why this was mostly an example of our use case we had. Generally the same core behavior remains. If any errors happens, either from the plugin or the parser it will report with `fatal:true` this happens for each file. So it's the same behavior as having a `sourceType:strict` instead of module.

Reply:
To avoid confusion, since ESLint itself doesn't read `tsconfig.json`.

> So it's not that typescript-eslint needs to handle this differently as it correctly outputs that it had a fatal error.

Yes, I believe that's the case, but it would be good to update the text to clarify that the desired exit code 2 at the end can't be achieved by making some changes in `@typescript-eslint/parser` as it already throws an error when this happens.

Reply:
Updated the docs and added this explanation about the behavior of tsconfig and `@typescript-eslint/parser`